src/AutomationTests/SystemUnitTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Abstractions;
+using Axe.Windows.Concretions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Axe.Windows.AutomationTests
+{
+    using Path = System.IO.Path;
+
+    [TestClass]
+    public class SystemUnitTests
+    {
+        private readonly ISystem _system = SystemFactory.CreateSystem();
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void DateTime_Matches()
```

Comment:
>DateTime_Matches [](start = 20, length = 16)

These unit tests are basically testing System methods. I'd suggest that what you really care about is that the class is mapping correctly to the underlying methods, so you can test these directly:

ISystem system = SystemFactory.CreateSystem();
Assert.AreEqual(System.DateTime, system.DateTime);
Assert.AreEqual(System.Environment, _system.Environment);
Assert.AreEqual(System.IO.Directory, _system.IO.Directory);

Reply:
I thought about that. But I think what I'm most interested in is that the system I've set up returns the actual system values. And besides, the way I've set it up also tests that the interfaces are set up as wel.

Reply:
The class you're testing is to map the interfaces to system classes. Your validation is relying on the behavior of those underlying classes to prove the mapping is correct. You probably wouldn't do this if the underlying class were some other class that you provided, since it violates the principle of only testing one class at a time. If you prove that the underlying class _is_ the system class, you don't need to prove that the underlying class _acts_ like the system class

---
In reply to: [298803943](https://github.com/microsoft/axe-windows/pull/79#discussion_r298803943) [](ancestors = 298803943)

Reply:
Rob and I discussed this, and there's no way to do what I'm describing

---
In reply to: [299598150](https://github.com/microsoft/axe-windows/pull/79#discussion_r299598150) [](ancestors = 299598150,298803943)